src/Api.Rest.Tests/Dtos/Data/SearchConditionParserTest.cs

```diff
@@ -0,0 +1,204 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2021                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Api.Rest.Tests.Dtos.Data
+{
+	#region usings
+
+	using System;
+	using System.Collections.Generic;
+	using FluentAssertions;
+	using NUnit.Framework;
+	using Zeiss.PiWeb.Api.Rest.Dtos.Data;
+
+	#endregion
+
+	[TestFixture]
+	public class SearchConditionParserTest
+	{
+		#region members
+
+		private static IEnumerable<string> _TestCases = new[]
```

Comment:
Please try to focus the tests on the edge cases, e.g. "123=[]", "123=[foo", "123=[fo]o", "123=[fo]o]", "123=[foo[]]", "123=[]foo]", ...